ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -910,6 +936,43 @@ func updateStatusToSuspendingIfNeeded(ctx context.Context, rayJob *rayv1.RayJob)
 	return true
 }
 
+func (r *RayJobReconciler) checkSubmitterAndUpdateStatusIfNeeded(ctx context.Context, rayJob *rayv1.RayJob, submissionMode rayv1.JobSubmissionMode) (shouldUpdate, isK8sJobFinished bool, err error) {
```

Comment:
Returning `isK8sJobFinished` is an abstraction leak for this function, especially if we want to unify all modes with a submitter. Instead, we can use `isSubmitterFinished` instead.

The value of `isSubmitterFinished` is based on the K8s Job (K8sJobMode) or the submitter sidecar container (SidecarMode)